CONTINUATION.md

```diff
@@ -0,0 +1,511 @@
+fastest resolved promise. However, it aggregates the exception values when all
+the promises are rejected.
+
+It is a merge point when exception values are aggregated.
+
+```js
+ctx.run('main', async () => {
+  try {
+    await Promise.any(_.times(5).map((it) => randomTask()))
+    ctx.get() // => (1)
+  } catch (e) {
+    ctx.get() // => (2)
+  }
+})
+```
+
+The value at site (1):
+
+- For `AsyncContext.Variable`, it is `main`,
+- For `ContinuationFlow`, it is the one caused the fulfillment, and discarding
+  leaf contexts of promises that may have been rejected.
+
+The value at site (2):
+
+- For `AsyncContext.Variable`, it is `main`,
+- For `ContinuationFlow`, the proposed solution didn't specify
+  conflicts-resolving yet.
+
+### Graft promises from outer scope
+
+The state of a resolved or rejected promise never changes, as in primitives or
+object identities. `Promise.prototype.then` creates a new promise from an
+existing one and automatically bubbles the return value and the exception
+to the new promise.
+
+`await` operations are meant to be practically similar to a
+`Promise.prototype.then`.
+
+By awaiting a promise created outside of the current context, the two
+executions are grafted together and there is a merge point on `await`.
+
+```js
+const gPromise = ctx.run('global', () => {
+  return Promise.resolve(1)
+})
+
+ctx.run('main', main)
+async function main() {
+  await gPromise
+  // -> global ---
+  //              \
+  // -> main   ----> (1)
+  ctx.get() // (1)
+}
+```
+
+The value at site (1):
+
+- For `AsyncContext.Variable`, it is `main`,
+- For `ContinuationFlow`, it is `global`, discarding the leaf context before
+  `await` operation.
+
+### Unhandled rejection
+
+Unhandled rejection events are tasks that is scheduled when
+[HostPromiseRejectionTracker](https://tc39.es/ecma262/#sec-host-promise-rejection-tracker)
+is invoked.
+
+The `PromiseRejectionEvent` instances of unhandled rejection events are emitted
+with the following properties:
+
+- `promise`: the promise which has no handler,
+- `reason`: the exception value.
+
+Intuitively, the context of the event should be relevant to the promise
+instance attached to the `PromiseRejectionEvent` instance.
+
+For a promise created with the `PromiseConstructor` or `Promise.withResolvers`,
+the promise can be rejected in a different context:
+
+```js
+let reject
+let p1 // (1)
+ctx.run('init', () => {
+  ;({ p1, reject } = Promise.withResolvers())
+})
+
+ctx.run('reject', () => {
+  reject('error message')
+})
+
+addEventListener('unhandledrejection', (event) => {
+  ctx.get() // (2)
+})
+```
+
+In this case, the `unhandledrejection` event will be dispatched with the promise
+instance `p1` and a string of `'error message'`, and the event is scheduled in
+the context of `'reject'`.
+
+For the two type of context variables with `unhandledrejection` event of `p1`:
+
+- For `AsyncContext.Variable`, the context is `'reject'`, where `p1` was rejected.
```

Comment:
I believe that the root context would be most consistent here.  I understand that we need access to the `reject` context and/or the `init` context, but if this is the _only_ case where we're not going with registration context, then it comes at a pretty high cost.  Instead, we should keep consistency and expose the causal context on the `Event`.

Reply:
Yeah, updated the example to use a more explicit form passing the event context with `event.asyncSnapshot`.